pygmt/clib/loading.py

```diff
@@ -84,6 +78,37 @@ def clib_name(os_name):
     return libname
 
 
+def clib_full_names(env=None):
+    """
+    Return the full path of GMT's shared library for the current OS.
+
+    Parameters
+    ----------
+    env : dict or None
+        A dictionary containing the environment variables. If ``None``, will
+        default to ``os.environ``.
+
+    Returns
+    -------
+    lib_fullnames: list of str
+        List of possible full names of GMT's shared library.
+
+    """
+    if env is None:
+        env = os.environ
+    libnames = clib_name(os_name=sys.platform)
+    libpath = env.get("GMT_LIBRARY_PATH", "")
+
+    lib_fullnames = [os.path.join(libpath, libname) for libname in libnames]
+    # Search for DLLs in PATH if GMT_LIBRARY_PATH is not defined [Windows only]
+    if not libpath and sys.platform == "win32":
+        for libname in libnames:
+            libfullpath = find_library(libname)
+            if libfullpath:
+                lib_fullnames.append(libfullpath)
```

Comment:
These lines were missing on my test coverage (since I'm on Linux) so I can't check this properly. But looking at the Azure Pipelines test on Windows, it seems okay.

Reply:
Is it OK to add a Windows-only test?

Reply:
The `find_library` function is cross platform, I wonder if we should remove the `sys.platform == 'win32'` requirement so that it's easier for Linux/MacOS users too. Comes at the risk of users picking up random `libgmt` libraries from elsewhere though.

Reply:
I'm not sure if `find_library` works on macOS. I can import pygmt correctly, which means `ctypes.CDLL("libgmt.dylib") works. However `find_library` returns nothing for me.
```python
In [1]: import pygmt

In [2]: from ctypes.util import find_library

In [3]: find_library("gmt")

In [4]: find_library("libgmt")

In [5]: find_library("libgmt.dylib")

In [6]: find_library("m")
Out[6]: '/usr/lib/libm.dylib'

In [7]: find_library("libm")
Out[7]: '/usr/lib/libm.dylib'

In [8]: find_library("libm.dylib")
Out[8]: '/usr/lib/libm.dylib'
```

Reply:
Could have sworn someone mentioned this before, but a quick scan through the old issues turn up nothing. Should be good to go then, but will keep it in the back of my mind :smile: